include/zephyr/net/hostname.h

```diff
@@ -27,6 +27,8 @@ extern "C" {
 	 (IS_ENABLED(CONFIG_NET_HOSTNAME_UNIQUE) ?	\
 	  sizeof("0011223344556677") - 1 : 0))
 
+#define NET_HOSTNAME_SIZE NET_HOSTNAME_MAX_LEN + 1
```

Comment:
It seems that this is causing CI issues in case `CONFIG_NET_HOSTNAME_ENABLE` is not set. We could do something like:
```c
#if defined(CONFIG_NET_HOSTNAME_ENABLE)
#define NET_HOSTNAME_SIZE NET_HOSTNAME_MAX_LEN + 1
#else
#define NET_HOSTNAME_SIZE 1
#endif
```
perhaps?

Reply:
So there was no need to change anything here?

Reply:
Ah, I was looking the first commit when commenting. This also means that the thing is not bisectable, could you move this change to the first commit.

Reply:
@jukkar That's a very good point. The change is moved.